ServiceMonitoringFunction/tst/com.amazon.vas.serviceordermonitoringlambda/util/TestDataBuilder.java

```diff
@@ -0,0 +1,185 @@
+    private final List<String> randomMerchantId = new ArrayList<>();
+    private final List<Integer> randomNumberOfJobs = new ArrayList<>();
+    private final int numberOfTestAggregatedJobs = 80;
+    private final long baseTime = 1592800200;
+    private final long timeGapBetweenConsecutiveTimeSlots = 3600;
+    public TestDataBuilder(){
+        generateRandomCity();
+        generateRandomSlotStartTime();
+        generateRandomMerchantId();
+        generateRandomTechnicianId();
+        generateRandomNumberOfJobs();
+
+    }
+    private void generateRandomNumberOfJobs(){
+        Random random = new Random();
+        for(int i=0;i<numberOfTestAggregatedJobs;i++)
+            randomNumberOfJobs.add(1+random.nextInt(10));
+    }
+    private void generateRandomCity(){
+        for(int i=0;i<numberOfTestAggregatedJobs;i++)
+            randomCities.add("City-"+i%10);
+    }
+    private void generateRandomMerchantId(){
+        Random random = new Random();
+        for(int i=0;i<numberOfTestAggregatedJobs;i++)
+            randomMerchantId.add("MerchantId-"+i%10);
+    }
+    private void generateRandomTechnicianId(){
+        Random random = new Random();
+        for(int i=0;i<numberOfTestAggregatedJobs;i++)
+            randomTechnicianId.add("TechnicianId-"+i%10);
+    }
+    private void generateRandomSlotStartTime(){
+        for(int i=0;i<numberOfTestAggregatedJobs;i++)
+            randomSlotStartTime.add(Long.toString(baseTime + timeGapBetweenConsecutiveTimeSlots * i/10));
+    }
+
+
+
+    public GetJobMetricsInputBO buildGetJobMetricsInputBO_withCityAndSlotStartTime(){
+        GetJobMetricsInputBO getJobMetricsInputBO = GetJobMetricsInputBO.builder().city(null)
+                .merchantId(null).serviceCategory(null)
+                .groupingCriteria(buildGroupingCriteria_cityAndSlotStartTime()).build();
+        return getJobMetricsInputBO;
+    }
+    public GetJobMetricsInput buildGetJobMetricsInput_withCityAndSlotStartTime(){
+        GetJobMetricsInput getJobMetricsInput = GetJobMetricsInput.builder().city(null)
+                .merchantId(null).serviceCategory(null)
+                .groupingCriteria(buildGroupingCriteria_cityAndSlotStartTime()).build();
+        return getJobMetricsInput;
+    }
+
+    public GetJobMetricsInputBO buildGetJobMetricsInputBO_withMerchantIdAndSlotStartTime(){
+        GetJobMetricsInputBO getJobMetricsInputBO = GetJobMetricsInputBO.builder().city(TEST_CITY)
+                .merchantId(null).serviceCategory(null)
+                .groupingCriteria(buildGroupingCriteria_merchantIdAndSlotStartTime()).build();
+        return getJobMetricsInputBO;
+    }
+
+    private List<String> buildGroupingCriteria_cityAndSlotStartTime() {
+        return new ArrayList<>(Arrays.asList(CITY, SLOT_START_TIME));
+    }
+    private List<String> buildGroupingCriteria_merchantIdAndSlotStartTime() {
+        return new ArrayList<>(Arrays.asList(MERCHANT_ID, SLOT_START_TIME));
+    }
```

Comment:
dont define such methods, just build these Arrays directly wherver you are using.